sdk/src/wallet/core/builder.rs

```diff
@@ -178,46 +175,78 @@ where
+        let mut verify_address = false;
+        let loaded_address = loaded_wallet_builder
+            .as_ref()
+            .and_then(|builder| builder.address.clone());
+
+        // May use a previously stored address if it wasn't provided
+        if let Some(address) = &self.address {
+            if let Some(loaded_address) = &loaded_address {
+                if address != loaded_address {
+                    return Err(crate::wallet::Error::WalletAddressMismatch(address.clone()));
+                }
+            } else {
+                verify_address = true;
+            }
+        } else {
+            self.address = loaded_address;
+        }
+
+        let loaded_bip_path = loaded_wallet_builder.as_ref().and_then(|builder| builder.bip_path);
 
         // May use a previously stored BIP path if it wasn't provided
         if let Some(bip_path) = self.bip_path {
-            if let Some(restored_bip_path) = restored_bip_path {
-                if bip_path != restored_bip_path {
+            if let Some(loaded_bip_path) = loaded_bip_path {
+                if bip_path != loaded_bip_path {
                     return Err(crate::wallet::Error::BipPathMismatch {
                         new_bip_path: Some(bip_path),
-                        old_bip_path: Some(restored_bip_path),
+                        old_bip_path: Some(loaded_bip_path),
                     });
                 }
+            } else {
+                verify_address = true;
             }
         } else {
-            self.bip_path = restored_bip_path;
+            self.bip_path = loaded_bip_path;
         }
 
+        // Create the node client.
+        let client = self
+            .client_options
+            .clone()
+            .ok_or(crate::wallet::Error::MissingParameter("client_options"))?
+            .finish()
+            .await?;
+
+        match (self.address.as_ref(), self.bip_path.as_ref()) {
+            (Some(address), Some(bip_path)) => {
+                if verify_address {
+                    // verify that the address is derived from the provided bip path.
+                    if let Some(backing_ed25519_address) = address.inner.backing_ed25519() {
+                        self.verify_ed25519_address(backing_ed25519_address, bip_path).await?;
+                    } else {
+                        return Err(crate::wallet::Error::InvalidParameter("address/bip_path mismatch"));
+                    }
+                }
+            }
+            (Some(address), None) => {}
+            (None, Some(bip_path)) => {
+                self.address.replace(Bech32Address::new(
+                    client.get_bech32_hrp().await?,
+                    self.generate_ed25519_address(bip_path).await?,
+                ));
+            }
+            (None, None) => {
+                return Err(crate::wallet::Error::MissingParameter("address, bip_path"));
```

Comment:
Does this make sense? Or `|` since we need only one
```suggestion
                return Err(crate::wallet::Error::MissingParameter("address or bip_path"));
```

Reply:
It's actually `and` here

Reply:
Why?

Reply:
We only need one?

Reply:
Ah yeah correct indeed, my bad